x/datadeal/types/message_deal.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+var _ sdk.Msg = &MsgCreateDeal{}
+
+func (m *MsgCreateDeal) Route() string {
+	return RouterKey
+}
+
+func (m *MsgCreateDeal) Type() string {
+	return "CreateDeal"
+}
+
+func (m *MsgCreateDeal) ValidateBasic() error {
+
+	return nil
+}
+
+func (m *MsgCreateDeal) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgCreateDeal) GetSigners() []sdk.AccAddress {
+	buyerAddress, err := sdk.AccAddressFromBech32(m.BuyerAddress)
```

Comment:
Oh my mistake.
Please change field `buyer_address` to `consumer_address` in `datadea/v2/tx.proto`

Reply:
In fact, I was curious about this. I'll change it!